### ISSUE TITLE
SQL cache table still incorrect; also document the index created

### ISSUE THREAD

**Issue by MV10** [dotnet/AspNetCore.Docs#20109]
This appears to be a regression in the docs. Closed issue https://github.com/dotnet/AspNetCore.Docs/issues/8383 asked about the `Id` column being shown as `NVARCHAR(900)` which led to PR https://github.com/dotnet/AspNetCore.Docs/pull/9038 to change it to 449, however the current documentation still shows 900 for some reason.

Compared to the source (below), there are other omissions like the collation declaration -- and I think the topic should also document the index the tool would create. In certain locked-down corporate scenarios, we can't just install arbitrary tools from the Internet.

https://github.com/dotnet/aspnetcore/blob/master/src/Tools/dotnet-sql-cache/src/SqlQueries.cs

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0d01025d-86b2-bb2d-86c1-ef0695bc98e3
* Version Independent ID: 5b9a6572-3671-0d54-b2ef-83d04f8b3e96
* Content: [Distributed caching in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/performance/caching/distributed?view=aspnetcore-3.1)
* Content Source: [aspnetcore/performance/caching/distributed.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/performance/caching/distributed.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-performance**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Indeed, something seems amiss here. The only image for this topic is ...

https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/performance/caching/distributed/_static/SqlServerCacheTable.png

... which matches the one that I placed on https://github.com/dotnet/AspNetCore.Docs/pull/9038 ...

https://github.com/dotnet/AspNetCore.Docs/blob/19f5a20859ed5c9f4fd5c019336395737310281f/aspnetcore/performance/caching/distributed/_static/SqlServerCacheTable.png

... and that's what's on the live branch at ...

https://github.com/dotnet/AspNetCore.Docs/blob/live/aspnetcore/performance/caching/distributed/_static/SqlServerCacheTable.png

... but the rendered topic is loading the **_old_** image ...

https://docs.microsoft.com/en-us/aspnet/core/performance/caching/distributed/_static/sqlservercachetable.png

Refreshing indicates that this it isn't a local caching problem.

@MV10 ... The `dotnet-sql-cache` tool is owned by Microsoft, so it's not fair to call it an "arbitrary tool." However, I know what you mean. In some places that I worked, **_every tool_**, including those from Microsoft, had to be approved. I agree that it would be nice to show the SQL commands, but idk if there's time for anyone from the team to add the content. We're all 🏃😅 for .NET 5 AFAIK.

**Comment by MV10** [comment]
> @MV10 ... The dotnet-sql-cache tool is owned by Microsoft, so it's not fair to call it an "arbitrary tool." However, I know what you mean. In some places that I worked, every tool, including those from Microsoft, had to be approved. I agree that it would be nice to show the SQL commands, but idk if there's time for anyone from the team to add the content. We're all 🏃😅 for .NET 5 AFAIK.

I probably should have worded it as "arbitrarily install tools" as we're very locked-down (big banking) and you know how the security guys get in corporate scenarios -- trust no one, etc. 

Anyway -- I certainly don't think this is urgent, I didn't expect an immediate fix, I just wanted to get it on The List. I imagine most people would just use the tool. I simply got to wondering about what indexes it created and wound up in this rabbit hole.

Looking forward to 5.0!!!

**Comment by mlsomers** [comment]
The documentation could be more useful by providing a SQL script like so:

```
CREATE TABLE [dbo].[MyCache](
	[Id] UNIQUEIDENTIFIER NOT NULL,
	[Value] [VARBINARY](MAX) NOT NULL,
	[ExpiresAtTime] [DATETIMEOFFSET](7) NOT NULL,
	[SlidingExpirationInSeconds] [BIGINT] NULL,
	[AbsoluteExpiration] [DATETIMEOFFSET](7) NULL,
PRIMARY KEY CLUSTERED 
(
	[Id] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = OFF, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
) ON [PRIMARY] TEXTIMAGE_ON [PRIMARY]
GO

CREATE NONCLUSTERED INDEX [Index_ExpiresAtTime] ON [dbo].[MyCache]
(
	[ExpiresAtTime] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, SORT_IN_TEMPDB = OFF, DROP_EXISTING = OFF, ONLINE = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
GO
```

I changed two things here:

- ID from `NVARCHAR(449)` to `UNIQUEIDENTIFIER`
- `ALLOW_PAGE_LOCKS = OFF` for the clustered index (not for `Index_ExpiresAtTime`)

The interface uses strings (seriously?) but after some testing I only ever see guids stored there, it seems to be working fine for me. The performance should be allot better than the default CI AI collation string comparison.
Could anyone point out why a decision was made for `NVARCHAR(449)` instead of `UNIQUEIDENTIFIER`?

It is great there is no EntityFramework dependency, though the [string consts here](https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/Caching/SqlServer/src/SqlQueries.cs) could have been stored procedures instead. We don't want to give Redis a head-start do we?

**Comment by postmeback** [comment]
> Could anyone point out why a decision was made for NVARCHAR(449) instead of UNIQUEIDENTIFIER?

Valid point.

**Comment by postmeback** [comment]
@Rick-Anderson can you guide me on this ?
I would love to contribute

**Comment by Rick-Anderson** [comment]
[Distributed Redis Cache](https://learn.microsoft.com/en-us/aspnet/core/performance/caching/distributed#distributed-redis-cache) is the recommended approach.
